utilities/include/alps/numeric/tensors/data_storage.hpp

```diff
@@ -7,11 +7,43 @@
 #ifndef ALPSCORE_GF_TENSORBASE_H
 #define ALPSCORE_GF_TENSORBASE_H
 
+#include <alps/numeric/tensors/mpi/mpi_shared_allocator.hpp>
 
 namespace alps {
   namespace numerics {
     namespace detail {
 
+      /**
+       * Simple data allocator
+       * @tparam T - datatype
+       */
+      template <typename T>
+      class simple_allocator {
+
+      public:
+        /**
+         * Allocate new array of requested size and return pointer to the first element
+         *
+         * @param size - size to allocate
+         * @return pointer to the first elemnt of allocated data array
+         */
+        T* allocate(size_t size) {
+          return new T[size];
+        }
+
+        /**
+         * release memory
+         * @param data
+         */
+        void deallocate(T* data) {
+          delete [] data;
+        }
+
+        void lock(){}
+        void release(){}
+        bool locked() {return true;}
```

Comment:
These methods are the central ingredient to your home-brew allocator protocol, so they are in dire need of documentation.  In particular, it is very surprising that locks are bound to the allocator, rather than the allocated memory, and this can only be explained by the fact that this is an abstraction of the MPI model.

Reply:
Because locks are connected with MPI window.

Reply:
I understand.  But I think it should have some documentation to that effect.